### DISCUSSION TITLE
Perhaps someone can assist?

### DISCUSSION THREAD

**Discussion by ArchieLinux** [Frederick888/external-editor-revived#74] [Q&A]
Greetings! Would love to use my nvim editor for composing/editing emails within Thunderbird (alternative would be to move to neomutt), but can't quite get this add-on to work after installation and selecting "nvim" + "konsole" options.

Here are the errors I'm getting
![Screenshot_20221205_072113](https://user-images.githubusercontent.com/119451251/205661795-9b7ae4d3-1b54-41fb-b50c-8cb4ed1a28be.png)

![Screenshot_20221205_072433](https://user-images.githubusercontent.com/119451251/205661850-5d082415-89ae-4d69-90cb-ac017eec473e.png)

.... essentially telling me that there's no such external-editor-revised extension... yet it's certainly been installed, and external editor menu choices are offered within Thunderbird.

Any ideas as to what I must be missing? Am excited to start using external editing possibilities!

**Comment by FernandoBasso** [comment]
Did you follow the extra steps mentioned in the wiki?

**Reply by ArchieLinux** [reply]
It's been a while ... but I thought I did before I gave up trying. 
I'll make time to have another go at it to see if I can get it working this time. Thanks!

**Reply by ArchieLinux** [reply]
Okay, tried again - a lot of additional activity but error messages: upon opening Thunderbird: "Messaging host disconnect"and upon efforts to utilize the external editor button: "Failed to talk to messaging host"

To clarify: I'm running Arch (EndeavourOS) and followed the wiki instructions again, specifically the link for Arch Linux. 

I installed `rust` and built the package successfully without errors reported, which created 'external-editor-revived 0.6.0-1' - which I then loaded as an add-on file within Thunderbird.

For other installations (e.g., Ubuntu) there are additional instructions for using different packages. I'm assuming since I'm not using Ubuntu that these don't apply - is this correct?

Thanks for any guidance you might provide.

**Comment by FernandoBasso** [comment]
On 2/15/23 17:54, ArchieLinux wrote:
> Okay, tried again - a lot of additional activity but error messages: 
> upon opening Thunderbird: "Messaging host disconnect"and upon efforts to 
> utilize the external editor button: "Failed to talk to messaging host"

I am using Arch Linux and downloaded the “Ubuntu” release package, and
ran the `external-editor-revived` program, which generated some output
which is supposed to be added to this file (on Linux at least):

     "$HOME/.mozilla/native-messaging-hosts/external_editor_revived.json"

Did you do these things?

**Comment by ArchieLinux** [comment]
@FernandoBasso Thank you kindly for your reply! ...

Yes, after downloading and building, I placed a copy of `external-editor-revived@tsundere.moe.xpi` into my `.mozilla/extensions` folder and a copy of `external_editor_revived.json` into my `.mozilla/native-messaging-hosts` folder. 

Quit Thunderbird, logged out, returned to receive same two error messages, which are as follows:

`ExtEditorR messaging host disconnected` when first restarting Thunderbird
`ExtEditorR failed to talk to messaging host` when I try to use the extension to edit a message.

Still stumped :thinking:
